# width shift applied

In [13]:
import numpy as np
import cv2
import os
import dlib
import matplotlib.pyplot as plt
from numpy import expand_dims
from keras.preprocessing.image import load_img
from keras.preprocessing.image import img_to_array
from keras.preprocessing.image import ImageDataGenerator
frames = []
output = []
for i in range(1,99):
    filename='../Images/oknot/okeynot ('+str(i)+').jpg'
    img = load_img(filename)
    data = img_to_array(img)
    samples = expand_dims(data, 0)
    datagen = ImageDataGenerator(width_shift_range=[-10,10])
    it = datagen.flow(samples, batch_size=1)
    for i in range(20):
        batch = it.next()
        image = batch[0].astype('uint8')
        #image_shape=image.shape
        gray_image = cv2.cvtColor(image,cv2.COLOR_RGB2GRAY)
        image_shape=gray_image.shape
        frames.append(gray_image.flatten())
        output.append(['notokey'])
X = np.array(frames)
y = np.array(output)

data = np.hstack([y, X])

f_name = "data.npy"

if os.path.exists(f_name):
    old = np.load(f_name)
    data = np.vstack([old, data])

np.save(f_name, data)
print(image_shape)

(48, 48)


In [14]:
data=np.load('data.npy')
data

array([['okey', '128', '128', ..., '127', '99', '49'],
       ['okey', '66', '69', ..., '82', '82', '82'],
       ['okey', '66', '69', ..., '82', '82', '82'],
       ...,
       ['notokey', '35', '15', ..., '124', '124', '124'],
       ['notokey', '35', '15', ..., '124', '124', '124'],
       ['notokey', '42', '42', ..., '108', '110', '112']], dtype='<U7')

In [15]:
X = data[:,1:].astype(int)
y = data[:,0]
print(X.shape)
print(y)

(14980, 2304)
['okey' 'okey' 'okey' ... 'notokey' 'notokey' 'notokey']


In [5]:
from matplotlib import pyplot
data = np.load('data.npy')
X = data[:,1:].astype(int)
y = data[:,0]
image=X[1452]
resizeimage = np.resize(image,(48,48))
print(resizeimage)

[[111 112 113 ...  21  21  21]
 [111 112 114 ...  52  52  52]
 [110 111 114 ...  45  45  45]
 ...
 [102  50  70 ...  69  69  69]
 [ 82  80  68 ...  67  67  67]
 [ 81  72  51 ...  62  62  62]]


# Height shift applied

In [16]:
import numpy as np
import cv2
import os
import dlib
import matplotlib.pyplot as plt
frames = []
output = []
for i in range(1,99):
    filename='../Images/oknot/okeynot ('+str(i)+').jpg'
    img = load_img(filename)
    data = img_to_array(img)
    samples = expand_dims(data, 0)
    datagen = ImageDataGenerator(height_shift_range=0.3)
    it = datagen.flow(samples, batch_size=1)
    for i in range(20):
        batch = it.next()
        image = batch[0].astype('uint8')
        #image_shape=image.shape
        gray_image = cv2.cvtColor(image,cv2.COLOR_RGB2GRAY)
        image_shape=gray_image.shape
        frames.append(gray_image.flatten())
        output.append(['notokey'])
X = np.array(frames)
y = np.array(output)

data = np.hstack([y, X])

f_name = "data.npy"

if os.path.exists(f_name):
    old = np.load(f_name)
    data = np.vstack([old, data])

np.save(f_name, data)
print(image_shape)
print(data.shape)

(48, 48)
(16940, 2305)


# Horizontal flip applied

In [17]:
import numpy as np
import cv2
import os
import dlib
import matplotlib.pyplot as plt
frames = []
output = []
for i in range(1,99):
    filename='../Images/oknot/okeynot ('+str(i)+').jpg'
    img = load_img(filename)
    data = img_to_array(img)
    samples = expand_dims(data, 0)
    datagen = ImageDataGenerator(horizontal_flip=True)
    it = datagen.flow(samples, batch_size=1)
    for i in range(20):
        batch = it.next()
        image = batch[0].astype('uint8')
        #image_shape=image.shape
        gray_image = cv2.cvtColor(image,cv2.COLOR_RGB2GRAY)
        image_shape=gray_image.shape
        frames.append(gray_image.flatten())
        output.append(['notokey'])
X = np.array(frames)
y = np.array(output)

data = np.hstack([y, X])

f_name = "data.npy"

if os.path.exists(f_name):
    old = np.load(f_name)
    data = np.vstack([old, data])

np.save(f_name, data)
print(image_shape)
print(data.shape)

(48, 48)
(18900, 2305)


# Vertical flip applied

In [18]:
import numpy as np
import cv2
import os
import dlib
import matplotlib.pyplot as plt
frames = []
output = []
for i in range(1,99):
    filename='../Images/oknot/okeynot ('+str(i)+').jpg'
    img = load_img(filename)
    data = img_to_array(img)
    samples = expand_dims(data, 0)
    datagen = ImageDataGenerator(vertical_flip=True)
    it = datagen.flow(samples, batch_size=1)
    for i in range(20):
        batch = it.next()
        image = batch[0].astype('uint8')
        #image_shape=image.shape
        gray_image = cv2.cvtColor(image,cv2.COLOR_RGB2GRAY)
        image_shape=gray_image.shape
        frames.append(gray_image.flatten())
        output.append(['notokey'])
X = np.array(frames)
y = np.array(output)

data = np.hstack([y, X])

f_name = "data.npy"

if os.path.exists(f_name):
    old = np.load(f_name)
    data = np.vstack([old, data])

np.save(f_name, data)
print(image_shape)
print(data.shape)

(48, 48)
(20860, 2305)


# Random Rotation applied

In [19]:
import numpy as np
import cv2
import os
import dlib
import matplotlib.pyplot as plt
frames = []
output = []
for i in range(1,99):
    filename='../Images/oknot/okeynot ('+str(i)+').jpg'
    img = load_img(filename)
    data = img_to_array(img)
    samples = expand_dims(data, 0)
    datagen = ImageDataGenerator(rotation_range=180)
    it = datagen.flow(samples, batch_size=1)
    for i in range(20):
        batch = it.next()
        image = batch[0].astype('uint8')
        #image_shape=image.shape
        gray_image = cv2.cvtColor(image,cv2.COLOR_RGB2GRAY)
        image_shape=gray_image.shape
        frames.append(gray_image.flatten())
        output.append(['notokey'])
X = np.array(frames)
y = np.array(output)

data = np.hstack([y, X])

f_name = "data.npy"

if os.path.exists(f_name):
    old = np.load(f_name)
    data = np.vstack([old, data])

np.save(f_name, data)
print(image_shape)
print(data.shape)

(48, 48)
(22820, 2305)


# Random brightness applied

In [20]:
import numpy as np
import cv2
import os
import dlib
import matplotlib.pyplot as plt
frames = []
output = []
for i in range(1,99):
    filename='../Images/oknot/okeynot ('+str(i)+').jpg'
    img = load_img(filename)
    data = img_to_array(img)
    samples = expand_dims(data, 0)
    datagen = ImageDataGenerator(brightness_range=[0.1,1.0])
    it = datagen.flow(samples, batch_size=1)
    for i in range(20):
        batch = it.next()
        image = batch[0].astype('uint8')
        #image_shape=image.shape
        gray_image = cv2.cvtColor(image,cv2.COLOR_RGB2GRAY)
        image_shape=gray_image.shape
        frames.append(gray_image.flatten())
        output.append(['notokey'])
X = np.array(frames)
y = np.array(output)

data = np.hstack([y, X])

f_name = "data.npy"

if os.path.exists(f_name):
    old = np.load(f_name)
    data = np.vstack([old, data])

np.save(f_name, data)
print(image_shape)
print(data.shape)

(48, 48)
(24780, 2305)


# Random Zoom applied

In [21]:
import numpy as np
import cv2
import os
import dlib
import matplotlib.pyplot as plt
frames = []
output = []
for i in range(1,99):
    filename='../Images/oknot/okeynot ('+str(i)+').jpg'
    img = load_img(filename)
    data = img_to_array(img)
    samples = expand_dims(data, 0)
    datagen = ImageDataGenerator(zoom_range=[0.3,1])
    it = datagen.flow(samples, batch_size=1)
    for i in range(20):
        batch = it.next()
        image = batch[0].astype('uint8')
        #image_shape=image.shape
        gray_image = cv2.cvtColor(image,cv2.COLOR_RGB2GRAY)
        image_shape=gray_image.shape
        frames.append(gray_image.flatten())
        output.append(['notokey'])
X = np.array(frames)
y = np.array(output)

data = np.hstack([y, X])

f_name = "data.npy"

if os.path.exists(f_name):
    old = np.load(f_name)
    data = np.vstack([old, data])

np.save(f_name, data)
print(image_shape)
print(data.shape)

(48, 48)
(26740, 2305)


# I have totally 26740 data

# I have 13020 okey images in 48X48 form 

# and 13720 not okey images in 48X48 form

In [22]:
data = np.load('data.npy')
data

array([['okey', '128', '128', ..., '127', '99', '49'],
       ['okey', '66', '69', ..., '82', '82', '82'],
       ['okey', '66', '69', ..., '82', '82', '82'],
       ...,
       ['notokey', '38', '19', ..., '131', '130', '133'],
       ['notokey', '46', '68', ..., '153', '155', '155'],
       ['notokey', '25', '32', ..., '132', '133', '138']], dtype='<U7')

In [25]:
data.shape

(26740, 2305)

In [38]:
x = data[:,0]
print(x)
print(x.shape)

['okey' 'okey' 'okey' ... 'notokey' 'notokey' 'notokey']
(26740,)


In [37]:
y = data[:,1:]
print(y)
print(y.shape)

[['128' '128' '128' ... '127' '99' '49']
 ['66' '69' '81' ... '82' '82' '82']
 ['66' '69' '81' ... '82' '82' '82']
 ...
 ['38' '19' '11' ... '131' '130' '133']
 ['46' '68' '83' ... '153' '155' '155']
 ['25' '32' '38' ... '132' '133' '138']]
(26740, 2304)
